libsolidity/analysis/TypeChecker.h

```diff
@@ -180,6 +180,8 @@ class TypeChecker: private ASTConstVisitor
 			return m_currentSourceUnit;
 	}
 
+	ArrayType const* m_arrayType = nullptr;
```

Comment:
TypeChecker should have as little state as possible, especially about things that are not syntax but expression related.